modules/python/examples/ukf-nonlinear-complex-example.py

```diff
@@ -0,0 +1,536 @@
+    return ColVector([range, relAngle])
+
+
+class vpLandmarksGrid:
+  """
+  @brief Class that represent a grid of landmarks that measure the distance and
+  relative orientation of the 4-wheel robot.
+  """
+  def __init__(self, landmarks: List[vpLandmarkMeasurements]):
+    """
+    @brief Construct a new vpLandmarksGrid object.
+
+    @param landmarks The list of landmarks forming the grid.
+    """
+    self.m_landmarks = landmarks # The list of landmarks forming the grid.
+
+  def state_to_measurement(self, chi: ColVector) -> ColVector:
+    """
+    @brief Convert the prior of the UKF into the measurement space.
+
+    @param chi The prior.
+    @return ColVector The prior expressed in the measurement space.
+    """
+    nbLandmarks = len(self.m_landmarks)
+    measurements = np.zeros(2*nbLandmarks)
+    for i in range (nbLandmarks):
+      landmarkMeas = self.m_landmarks[i].state_to_measurement(chi)
+      measurements[2*i] = landmarkMeas[0]
+      measurements[(2*i) + 1] = landmarkMeas[1]
+    return ColVector(measurements)
+
+  def measureGT(self, pos: ColVector) -> ColVector:
+    """
+    @brief Perfect measurement from each landmark of the range and relative orientation of the robot
+    located at pos.
+
+    @param pos The actual position of the robot (pos[0]: x, pos[1]: y, pos[2] = heading.
+    @return ColVector n x ([0] the range [1] the relative orientation of the robot), where
+    n is the number of landmarks.
+    """
+    nbLandmarks = len(self.m_landmarks)
+    measurements = np.zeros(2*nbLandmarks)
+    for i in range(nbLandmarks):
+      landmarkMeas = self.m_landmarks[i].measureGT(pos)
+      measurements[2*i] = landmarkMeas[0]
+      measurements[(2*i) + 1] = landmarkMeas[1]
+    return ColVector(measurements)
+
+  def measureWithNoise(self, pos: ColVector) -> ColVector:
+    """
+    @brief Noisy measurement from each landmark of the range and relative orientation that
+    correspond to pos.
+
+    @param[in] pos The actual position of the robot (pos[0]: x ; pos[1] = y ; pos[2] = heading).
+    @return ColVector n x ([0] the range [1] the relative orientation of the robot), where
+    n is the number of landmarks.
+    """
+    nbLandmarks = len(self.m_landmarks)
+    measurements = np.zeros(2*nbLandmarks)
+    for i in range(nbLandmarks):
+      landmarkMeas = self.m_landmarks[i].measureWithNoise(pos)
+      measurements[2*i] = landmarkMeas[0]
+      measurements[(2*i) + 1] = landmarkMeas[1]
+    return ColVector(measurements)
+
+if __name__ == '__main__':
+  dt = 0.1 # Period of 0.1s
+  step = 1. # Number of update of the robot position between two UKF filtering
+  sigmaRange = 0.3 # Standard deviation of the range measurement: 0.3m
+  sigmaBearing = Math.rad(0.5) # Standard deviation of the bearing angle: 0.5deg
+  wheelbase = 0.5 # Wheelbase of 0.5m
+  processVariance = 0.0001
+  landmarks = [ vpLandmarkMeasurements(5, 10, sigmaRange, sigmaBearing)
```

Comment:
You could replace this expression with a list comprehension:
positions = [(5, 10), (10, 5)...]
landmarks = [LandmarkMeasurements(x, y, sigma_range, sigma_bearing) for x,y in positions]